WordPress/Classes/Login/SelfHostedSiteAuthenticator.swift

```diff
@@ -79,11 +100,40 @@ final actor SelfHostedSiteAuthenticator {
         let timestamp = ISO8601DateFormatter.string(from: .now, timeZone: .current, formatOptions: .withInternetDateTime)
         let appNameValue = "\(appName) - \(deviceName) (\(timestamp))"
 
-        return try await internalClient.login(
-            site: site,
-            appName: appNameValue,
-            appId: appId
-        )
+        let loginURL = try await internalClient.loginURL(forSite: site, application: .init(id: appId, name: appNameValue, callbackUrl: SelfHostedSiteAuthenticator.callbackURL.absoluteString))
+        let callback = try await authenticate(url: loginURL, callbackURL: SelfHostedSiteAuthenticator.callbackURL, from: anchor)
+        return try internalClient.credentials(from: callback)
+    }
+
+    @MainActor
+    func authenticate(url: URL, callbackURL: URL, from anchor: ASPresentationAnchor) async throws -> URL {
+        let provider = WebAuthenticationPresentationAnchorProvider(anchor: anchor)
```

Comment:
You can actually just pass in `ASPresentationAnchor()` here – no need to pass something down the whole hierarchy

Reply:
There was [this crash](https://github.com/wordpress-mobile/WordPress-iOS/pull/23777#issuecomment-2462495593), which [I couldn't reproduce](https://github.com/wordpress-mobile/WordPress-iOS/pull/23777#issuecomment-2463222787). I suspect the crash was due to using a new window instance. So I try to pass an existing window down to the web authentication session instance, instead of creating a new one.